apps/site/src/lib/components/forms/shared/ApplyConfirmation/ConfirmationDetails.tsx

```diff
@@ -29,7 +29,11 @@ export default async function ConfirmationDetails({
 			<Button
 				className="text-2xl"
 				text={isLoggedIn ? "Proceed to Application" : "Log in to Apply"}
-				href={isLoggedIn ? `${continueHREF}?prefaceAccepted=true` : "/login"}
+				href={
+					isLoggedIn
+						? `${applicationURL}?prefaceAccepted=true`
+						: `/login?return_to=%2F${applicationURL.substring(1)}`
```

Comment:
Nitpick: using URLSearchParams would be safer and more understandable, e.g.
```suggestion
						: `/login?${new URLSearchParams({ return_to: applicationUrl })}`
```